calico_kubernetes/calico_kubernetes.py

```diff
@@ -0,0 +1,353 @@
+import os
+import sys
+from subprocess import check_output, CalledProcessError, check_call
+import requests
+import sh
+
+# Append to existing env, to avoid losing PATH etc.
+# Need to edit the path here since calicoctl loads client on import.
+ETCD_AUTHORITY_ENV = "ETCD_AUTHORITY"
+if ETCD_AUTHORITY_ENV not in os.environ:
+    os.environ[ETCD_AUTHORITY_ENV] = 'kubernetes-master:6666'
+print("Using ETCD_AUTHORITY=%s" % os.environ[ETCD_AUTHORITY_ENV])
+
+from pycalico.datastore import IF_PREFIX
+from pycalico.util import generate_cali_interface_name, get_host_ips
+
+CALICOCTL_PATH = os.environ.get('CALICOCTL_PATH', '/usr/bin/calicoctl')
+print("Using CALICOCTL_PATH=%s" % CALICOCTL_PATH)
+
+KUBE_API_ROOT = os.environ.get('KUBE_API_ROOT',
+                               'http://kubernetes-master:8080/api/v1/')
+print("Using KUBE_API_ROOT=%s" % KUBE_API_ROOT)
+
+
+class NetworkPlugin(object):
+    def __init__(self):
+        self.pod_name = None
+        self.docker_id = None
+        self.calicoctl = sh.Command(CALICOCTL_PATH).bake(_env=os.environ)
+
+    def create(self, pod_name, docker_id):
+        """"Create a pod."""
+        # Calicoctl does not support the '-' character in iptables rule names.
+        # TODO: fix Felix to support '-' characters.
+        self.pod_name = pod_name
+        self.docker_id = docker_id
+
+        print('Configuring docker container %s' % self.docker_id)
+
+        try:
+            endpoint = self._configure_interface()
+            self._configure_profile(endpoint)
+        except CalledProcessError as e:
+            print('Error code %d creating pod networking: %s\n%s' % (
+                e.returncode, e.output, e))
+            sys.exit(1)
+
+    def delete(self, pod_name, docker_id):
+        """Cleanup after a pod."""
+        self.pod_name = pod_name
+        self.docker_id = docker_id
+
+        print('Deleting container %s with profile %s' % 
+            (self.pod_name, self.docker_id))
+
+        # Remove the profile for the workload.
+        self.calicoctl('container', 'remove', self.docker_id)
+        self.calicoctl('profile', 'remove', self.pod_name)
+
+    def _configure_interface(self):
+        """Configure the Calico interface for a pod.
+
+        This involves the following steps:
+        1) Determine the IP that docker assigned to the interface inside the
+           container
+        2) Delete the docker-assigned veth pair that's attached to the docker
+           bridge
+        3) Create a new calico veth pair, using the docker-assigned IP for the
+           end in the container's namespace
+        4) Assign the node's IP to the host end of the veth pair (required for
+           compatibility with kube-proxy REDIRECT iptables rules).
+        """
+        container_ip = self._read_docker_ip()
+        self._delete_docker_interface()
+        print('Configuring Calico network interface')
+        ep = self.calicoctl('container', 'add', self.docker_id, container_ip, 'eth0')
```

Comment:
Don't think this will work the way you expect it to - "calicoctl container add" from the command line is not going to return an Enpoint object like calling container_add( ... ) did.

I'm not sure what the best solution is here, since we want to be replacing this.  Let's chat about this one.